### PR TITLE
Update operator deps and allow istio retries

### LINKED ISSUE #1402: Flakiness on integration tests
## Context

Sometimes integration tests seem to behave flaky and raise false errors. We have noticed two different cases where this usually happens:

1. `503` or `504` errors on one or two of the tests in [testing/scripts/test_rolling_updates.py](https://github.com/SeldonIO/seldon-core/blob/master/testing/scripts/test_rolling_updates.py). An example can be seen on the [log output for PR #1400](https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1400/3.log). This has happened in the past either due to issues with Ambassador / Istio falling over to the new deployment (actual red flag) or due to spamming the `SeldonDeployment` pod with too many requests.

  https://github.com/SeldonIO/seldon-core/blob/f21fa8e8f16baef2455158eb67ae10c826c0f4ec/jenkins-x/logs/SeldonIO/seldon-core/PR-1400/3.log#L18139-L18170

2. An error installing Istio. In particular, a timeout waiting for the `istio-ingressgateway` pod to come up. Instead of failing early, the tests continue and this in turn causes (almost) all of them to fail. An example can be seen on the [log output for PR #1382](https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1382/4.log). This has happened in the past whenever the cluster ran out of ephemeral storage (e.g. #1322).

https://github.com/SeldonIO/seldon-core/blob/f21fa8e8f16baef2455158eb67ae10c826c0f4ec/jenkins-x/logs/SeldonIO/seldon-core/PR-1382/4.log#L340-L344

On both cases they seem to be intermittent. 

## Workaroud

While they are not fixed properly, we can verify if it's a red flag by running them locally (and adding a note about that on the PR for the reviewers).

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes #1402 

 * Update operator core libraries to 1.17.2
 * Allow istio retries settings
 * Update rolling update tests to stop flakiness by handling 503s with retries in Ambassador and istio

Todo:

 * Update docs

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Sat Mar  7 14:41:32 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=1

**Comment by seldondev** [comment]
Sat Mar  7 14:41:42 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=2

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 59b0d17348393ddd196324a156ac7157de242a52 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Sat Mar  7 15:00:32 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=4

**Comment by seldondev** [comment]
Sat Mar  7 15:00:42 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=3

**Comment by seldondev** [comment]
Sat Mar  7 15:02:49 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=5

**Comment by seldondev** [comment]
Sat Mar  7 15:02:59 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=6

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | a4cffa0565ca608f1f20d2fa39e7775f8bbd2cb4 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 5a4c2cd56ab7338d20b222469603a4b0570c7325 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Sat Mar  7 15:45:40 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=7

**Comment by seldondev** [comment]
Sat Mar  7 15:45:42 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=8

**Comment by seldondev** [comment]
Sat Mar  7 18:45:11 UTC 2020
The logs for [integration] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=9

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 760a1e4bf46adb446ca3683a719b96a6037667c3 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Sat Mar  7 23:07:24 UTC 2020
The logs for [integration] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=10

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 760a1e4bf46adb446ca3683a719b96a6037667c3 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Sun Mar  8 09:25:13 UTC 2020
The logs for [pr-build] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=11

**Comment by seldondev** [comment]
Sun Mar  8 09:25:24 UTC 2020
The logs for [lint] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=12

**Comment by seldondev** [comment]
Sun Mar  8 09:39:41 UTC 2020
The logs for [integration] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=13

File: operator/controllers/suite_test.go [hunk 1]
```diff
@@ -125,7 +126,8 @@ var _ = BeforeSuite(func(done Done) {
 	//apiServerFlags = append(apiServerFlags, "--admission-control=MutatingAdmissionWebhook")
 
 	testEnv = &envtest.Environment{
-		CRDDirectoryPaths: []string{filepath.Join("..", "config", "crd", "bases"),
+		CRDDirectoryPaths: []string{
+			//filepath.Join("..", "config", "crd", "bases"),
```
Comment (adriangonz) [inline operator/controllers/suite_test.go]: I can see you added the `SeldonDeployment` CRD to `../testing` as well. For future changes to the CRDs, would `kubebuilder` also write to that location?

File: testing/scripts/conftest.py [hunk 2]
```diff
@@ -32,6 +32,8 @@ def namespace(request):
 
     test_name = request.node.name
     namespace = clean_string(test_name)
+    if len(namespace) > 63:
+        namespace = namespace[0:63]
```
Comment (adriangonz) [inline testing/scripts/conftest.py]: Good spot!

File: testing/scripts/seldon_e2e_utils.py [hunk 3]
```diff
@@ -49,6 +60,31 @@ def wait_for_shutdown(deployment_name, namespace, timeout="10m"):
...
+        logging.warning(
+            f"Unsuccessful wait command but retrying for SeldonDeployment pod {sdep_name}"
+        )
+        time.sleep(sleep)
+    assert ret.returncode == 0, "Failed to get  pod names: non-zero return code"
+    data = json.loads(ret.stdout)
+    pod_names = []
+    for item in data["items"]:
+        pod_names.append(item["metadata"]["name"])
+    logging.warning(
```
Comment (adriangonz) [inline testing/scripts/seldon_e2e_utils.py]: Should we change the `warning`s to `info`s?

File: testing/scripts/test_rolling_updates.py [hunk 4]
```diff
@@ -226,21 +226,25 @@ def test_rolling_update10(self, namespace, api_gateway):
...
+        ("graph1.json", "graph2.json", True),  # New image version
+        (
+            "graph1.json",
+            "graph3.json",
+            True,
+        ),  # New image version and new name of container
+        ("graph1.json", "graph4.json", True),  # New resource request but same image
+        ("graph1.json", "graph5.json", True),  # Update with multi-deployment new model
+        ("graph1.json", "graph8.json", True),  # From v1alpha2 to v1
+        ("graph7.json", "graph8.json", False),  # From v1alpha3 to v1
```
Comment (adriangonz) [inline testing/scripts/test_rolling_updates.py]: Does `change=False` mean that the current pod doesn't get deleted? How is the update from `v1alpha3` to `v1` happening?

File: operator/controllers/suite_test.go (hunk 1 above)
Comment (cliveseldon) [reply operator/controllers/suite_test.go]: yes - not done automatically. We need the kustomize created version of the CRD. Will add to Makefile to generate.

File: testing/scripts/test_rolling_updates.py (hunk 4 above)
Comment (cliveseldon) [reply testing/scripts/test_rolling_updates.py]: As I see it as there is no change between these resources it gets changed automatically by the k8s api to the stored version so there is literally no difference here.

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | a0d11a390dbe63a0b16728315cf19399175bafd1 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | a0d11a390dbe63a0b16728315cf19399175bafd1 | [link]() | `/test lint`
pr-build | a0d11a390dbe63a0b16728315cf19399175bafd1 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: operator/controllers/suite_test.go (hunk 1 above)
Comment (cliveseldon) [reply operator/controllers/suite_test.go]: done

File: testing/scripts/seldon_e2e_utils.py (hunk 3 above)
Comment (cliveseldon) [reply testing/scripts/seldon_e2e_utils.py]: done

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | a0d11a390dbe63a0b16728315cf19399175bafd1 | [link]() | `/test lint`
pr-build | 1e85eab9fa5ef1c36ec6da0e6265ac44a17dff3e | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 1e85eab9fa5ef1c36ec6da0e6265ac44a17dff3e | [link]() | `/test this`
lint | 1e85eab9fa5ef1c36ec6da0e6265ac44a17dff3e | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Mar  9 15:38:51 UTC 2020
The logs for [lint] [19] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/19.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=19

**Comment by seldondev** [comment]
Mon Mar  9 15:39:00 UTC 2020
The logs for [pr-build] [18] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/18.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=18

File: testing/scripts/test_rolling_updates.py (hunk 4 above)
Comment (adriangonz) [reply testing/scripts/test_rolling_updates.py]: Got it! That makes sense.

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1518#issuecomment-596650451" title="Approved">adriangonz</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [adriangonz]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Mar  9 16:57:33 UTC 2020
The logs for [integration] [20] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/20.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=20

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | be4431d6c02e52fbbe31bef1bae03e359b4591be | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Mar  9 19:46:39 UTC 2020
The logs for [integration] [21] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/21.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=21

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | be4431d6c02e52fbbe31bef1bae03e359b4591be | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Mar 10 15:04:13 UTC 2020
The logs for [pr-build] [22] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/22.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=22

**Comment by seldondev** [comment]
Tue Mar 10 15:04:20 UTC 2020
The logs for [lint] [23] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/23.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=23

**Comment by seldondev** [comment]
Tue Mar 10 15:05:09 UTC 2020
The logs for [lint] [25] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/25.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=25

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | be4431d6c02e52fbbe31bef1bae03e359b4591be | [link]() | `/test integration`
lint | e4fccb1b3f842a2aa9b38dbfd935804b69733274 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Mar 10 15:06:20 UTC 2020
The logs for [pr-build] [24] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/24.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=24

**Comment by seldondev** [comment]
Tue Mar 10 15:09:06 UTC 2020
The logs for [integration] [26] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1518/26.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1518 --build=26

**Pull request merged**